server/src/main/java/org/elasticsearch/common/lucene/uid/VersionsAndSeqNoResolver.java

```diff
@@ -34,7 +34,8 @@ public final class VersionsAndSeqNoResolver {
         }
     };
 
-    private static PerThreadIDVersionAndSeqNoLookup[] getLookupState(IndexReader reader, String uidField) throws IOException {
+    private static PerThreadIDVersionAndSeqNoLookup[] getLookupState(IndexReader reader, String uidField, boolean loadTimestampRange)
```

Comment:
Can we have some form of protection against the case when `getLookupState` would be called with `loadTimestampRange`=false then `loadTimestampRange`=true, as in that case we'd retrieve the entry from the threadlocal which wouldn't have a timestamp range?